### PR TITLE
V22F-642 Update replay blackbox 3

### PR DESCRIPTION
**Pull request by noutheesink**
Replay blackbox reads binary files. Makes 4 decisions.

### CODE REVIEW

**Review by KungFuDonkey — APPROVED** [review]

File: SDALib-CPP/SDAReplay/ReplayDriver.h [hunk 1]
```diff
@@ -6,29 +6,50 @@
...
+    std::cout << "Reading " << sizeof(TYPE) << " bits" << std::endl;
+    return p_in.read(reinterpret_cast<char*>(&p_b.T), sizeof(TYPE));
+}
+
+template <typename TYPE>
+static inline std::ostream& operator<<(std::ostream& p_out, Bits<TYPE&> const p_b)
+{
+    // reinterpret_cast is for pointer conversion
+    // static_cast is for compatible pointer conversion
+    return p_out.write(reinterpret_cast<const char*>(&(p_b.t)), sizeof(TYPE));
```
Comment (shoekzema) [inline SDALib-CPP/SDAReplay/ReplayDriver.h:39]: any reason why this comment is here?

**Review by shoekzema — CHANGES_REQUESTED** [review]
Functions ReplayDriver(), InitAI() and UpdateAI() are implemented in a header file. Should be in a .cpp file. They are also missing documentation.

**Comment by KungFuDonkey** [comment]
> Functions ReplayDriver(), InitAI() and UpdateAI() are implemented in a header file. Should be in a .cpp file. They are also missing documentation.

That is not entirely true as we didn't do that in most of the files on SDALib, we can make this a bug and convert the entire library later

File: SDALib-CPP/SDAReplay/ReplayDriver.h (hunk 1 above)
Comment (noutheesink) [reply SDALib-CPP/SDAReplay/ReplayDriver.h:39]: Not really

**Review by shoekzema — APPROVED** [review]

**Pull request merged**